apps/api/src/domain/models/bibliographic-reference/common/commands/register-digital-representation-of-bibiliographic-citation/register-digital-representation-of-bibliographic-citation.command.integration.spec.ts

```diff
@@ -0,0 +1,192 @@
+                        dummySystemUserId
+                    );
+                },
+            });
+        });
+    });
+
+    describe(`when the command is invalid`, () => {
+        describe(`when the bibliographic citation does not exist`, () => {
+            it(`should fail with the expected errors`, async () => {
+                await assertCommandError(assertionHelperDependencies, {
+                    systemUserId: dummySystemUserId,
+                    seedInitialState: async () => {
+                        await testRepositoryProvider.addFullSnapshot(
+                            new DeluxeInMemoryStore({
+                                [ResourceType.digitalText]: [existingDigitalText],
+                                // empty
+                                [ResourceType.bibliographicReference]: [],
+                            }).fetchFullSnapshotInLegacyFormat()
+                        );
+                    },
+                    buildCommandFSA: buildValidCommandFSA,
+                    checkError: (error) => {
+                        assertErrorAsExpected(
+                            error,
+                            new CommandExecutionError([
+                                new AggregateNotFoundError(
+                                    existingBibliographicReference.getCompositeIdentifier()
+                                ),
+                            ])
+                        );
+                    },
+                });
+            });
+        });
+
+        describe(`when the digital representation resource (digital text) does not exist`, () => {
+            it(`should fail with the expected error`, async () => {
+                await assertCommandError(assertionHelperDependencies, {
+                    systemUserId: dummySystemUserId,
+                    seedInitialState: async () => {
+                        await testRepositoryProvider.addFullSnapshot(
+                            new DeluxeInMemoryStore({
+                                [ResourceType.bibliographicReference]: [
+                                    existingBibliographicReference,
+                                ],
+                            }).fetchFullSnapshotInLegacyFormat()
+                        );
+                    },
+                    buildCommandFSA: buildValidCommandFSA,
+                    checkError: (error) => {
+                        assertErrorAsExpected(
+                            error,
+                            new CommandExecutionError([
+                                new InvalidExternalReferenceByAggregateError(
+                                    existingBibliographicReference.getCompositeIdentifier(),
+                                    [existingDigitalText.getCompositeIdentifier()]
+                                ),
+                            ])
+                        );
+                    },
+                });
+            });
+        });
+    });
```

Comment:
What about when the digital representation is already in use by a different bibliographic citation?

Reply:
Validation \ test added.